src/use-swr.ts

```diff
@@ -122,6 +122,7 @@ const mutate: mutateInterface = async (
   if (typeof data !== 'undefined') {
     // update cached data, avoid notifying from the cache
     cache.set(key, data, false)
+    cache.set(keyErr, error, false)
```

Comment:
if data is undefined but error is not?

```js
if (typeof error !== 'undefined') {
   cache.set(keyErr, error, false)
}
```

Reply:
I don't think it might be necessary since this will update `keyErr` to undefined to clear the key unless there was an error, in that case should update to that new error 🤔

Reply:
yea we might not need to have this condition inside `if` scope, but I think we might want to move cache set outside of condition `if (typeof data !== 'undefined')`?

if the data is undefined but error is not, then we still didn't update the `keyError` in cache.

I created a test case in #481 , it won't pass if we update error cache when data is available

Reply:
Yes I agree @huozhi. Consider the following case:

```js
mutate(key, post(API))
```

Where post is an async function and it throws. In that case, we will expect the cache to be updated:
- data → cleared
- error → set

Reply:
It should be like this right?
```javascript
if (typeof data !== 'undefined') {
  // update cached data, avoid notifying from the cache
  cache.set(key, data, false)
}
cache.set(keyErr, error, false)
```

Reply:
Yep!

Reply:
@shuding thanks! @huozhi Thanks for creating a test case!